### PR TITLE
Interfacing stateDB cache

### PR DESCRIPTION
**Pull request by aidan-kwon**
## Proposed changes

- This PR Interfaces stateDB cache
- A new implementation of stateDB cache will be introduced

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: storage/statedb/cache.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package statedb
+
+import "github.com/VictoriaMetrics/fastcache"
+
+// Cache interface the cache of stateDB
+type Cache interface {
+	Set(k, v []byte)
+	Get(dst, k []byte) []byte
+	HasGet(dst, k []byte) ([]byte, bool)
+	UpdateStats(s *fastcache.Stats)
```
Comment (ehnuje) [inline storage/statedb/cache.go:10]: These interface methods can be changed into more general ones like `Get(key []byte)`, `Has(key []byte)` ... 
Also, `UpdateStats` depends on the implementation of `fastcache`.

**Comment by winnie-byun** [comment]
I still see a `fastcache.Stats` in `UpdateMetricNodes()` and `warmUpLoop()`. Wouldn't this be interfaced, too?

File: storage/statedb/cache.go (hunk 1 above)
Comment (aidan-kwon) [reply storage/statedb/cache.go]: Generalized them.

**Comment by aidan-kwon** [comment]
> I still see a `fastcache.Stats` in `UpdateMetricNodes()` and `warmUpLoop()`. Wouldn't this be interfaced, too?

@winnie-byun I was intended at the first time to minimize change. Anyway, it is changed now.

File: storage/statedb/cache_fastcache.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package statedb
```
Comment (KimKyungup) [inline storage/statedb/cache_fastcache.go:1]: How about changing this file to `localcache.go`?

**Review by KimKyungup — APPROVED** [review]

**Review by winnie-byun — APPROVED** [review]

File: storage/statedb/cache_fastcache.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import "github.com/VictoriaMetrics/fastcache"
+
+type LocalCache struct {
```
Comment (ehnuje) [inline storage/statedb/cache_fastcache.go:5]: The name `LocalCache` suggests that there is a `RemoteCache` struct, but there isn't. 
Also, `LocalCache` sounds like an interface, not a struct. 
How about just wrapping `fastcache.cache` as `statedb.FastCache`?

Comment (aidan-kwon) [reply storage/statedb/cache_fastcache.go:5]: 1. RemoteCache will be introduced later soon.
2. When `fastcache.cache` is wrapped to `statedb.FastCache`, type assertion is needed in every method of `statedb.FastCache` to use methods of `fastcache.cache`. In that case, newly introduced interfaces should returns an error also.

Comment (aidan-kwon) [reply storage/statedb/cache_fastcache.go:5]: I am gonna try to find a new name replacing `LocalCache` later. If you have any idea, please let me know.

Comment (ehnuje) [reply storage/statedb/cache_fastcache.go:5]: What I meant for "wrapping fastcache.cache as statedb.FastCache" is to change the name of current `LocalCache` to `FastCache`, not changing the current interfaces. Let's proceed as what you intended and check if additional change is needed later. I think it is good for now.

**Review by ehnuje — APPROVED** [review]

File: storage/statedb/cache_fastcache.go (hunk 3 above)
Comment (aidan-kwon) [reply storage/statedb/cache_fastcache.go:5]: OK. I will consider your comment in the next PR.

**Pull request merged**